spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -119,7 +121,24 @@ public WriteBuilder overwrite(Filter[] filters) {
         !overwriteFiles, "Cannot overwrite individual files and using filters");
     Preconditions.checkState(rewrittenFileSetId == null, "Cannot overwrite and rewrite");
 
-    this.overwriteExpr = SparkFilters.convert(filters);
+    boolean caseSensitive = Boolean.parseBoolean(spark.conf().get("spark.sql.caseSensitive"));
+    overwriteExpr = Expressions.alwaysTrue();
+    for (Filter filter : filters) {
```

Comment:
Can you update the `convert(Filter[])` method rather than moving so much of the logic here?